packages/app/src/cli/models/app/app.ts

```diff
@@ -6,17 +6,28 @@ import {getDependencies, PackageManager, readAndParsePackageJson} from '@shopify
 import {fileRealPath, findPathUp} from '@shopify/cli-kit/node/fs'
 import {joinPath, dirname} from '@shopify/cli-kit/node/path'
 
-export const AppConfigurationSchema = zod.object({
+export const legacyAppSchema = zod.object({
+  extension_directories: zod.array(zod.string()).optional(),
+  web_directories: zod.array(zod.string()).optional(),
+  scopes: zod.string().default(''),
+})
+
+export const appSchema = zod.object({
   extension_directories: zod.array(zod.string()).optional(),
   web_directories: zod.array(zod.string()).optional(),
   name: zod.string().optional(),
   client_id: zod.string().optional(),
-  scopes: zod.string().default(''),
   application_url: zod.string().optional(),
   redirect_url_allowlist: zod.array(zod.string()).optional(),
   requested_access_scopes: zod.array(zod.string()).optional(),
 })
 
+export const AppConfigurationSchema = zod.union([legacyAppSchema, appSchema])
+
+export function isType<T extends zod.ZodTypeAny>(schema: T, item: unknown): item is zod.infer<T> {
```

Comment:
we could also define more specific helpers to match a particular type, if preferred. advantage to that is it's less imports in the implementation since you won't need to implement the particular schema

Reply:
hmm yeah, this function is asking the question "is this item this specific type" but maybe we want an updated or separate function for "treat it as new config or old config, throw if it's invalid"... or something that generally considers validity

Reply:
our three branches for app schema will be "is it legacy" "is it new" or "is it borked" and we might want something easy to clearly distinguish those, vs. (or in addition to) a helper of "is it this type" <-- is this what you're getting at?